var/spack/repos/builtin/packages/cloverleaf/package.py

```diff
@@ -92,7 +92,7 @@ def cmake_args(self):
             if "cuda_arch" in spec.variants:
                 cuda_arch = spec.variants["cuda_arch"].value[0]
                 args.append(self.define("OFFLOAD", "NVIDIA:{0}".format(cuda_arch)))
-        elif "+rocm" in spec:
+        elif "+rocm" in spec and "~raja" in spec:
```

Comment:
What is the spec for when RAJA (AMD) is enabled? It should be simply `cloverleaf@main +raja` which would not have a 'rocm' variant in the main spec.